### DISCUSSION TITLE
How to emitt logs attached to the CalcJobNode while parsing?

### DISCUSSION THREAD

**Discussion by zhubonan** [aiidateam/aiida-core#4906] [Q&A] [answered]
When parsing output files of a calculation it can be useful to emit logs that include detailed exceptions traceback should they occur, and the `Parser` class does have a `logger` property. However, it seems that the logs are not stored in the database but only printed to the console (hence appear in the daemon logs). In contrast, the `report` method within a `WorkChain` does emit and log that is stored in the database. 

Is it possible to make the `Parser.logger` do the same?

**Comment by sphuber** [comment, marked as answer]
Yes this should be possible and I think this already happens. The `Parser` class creates a logger using the logger adapter
```
class Parser(ABC):
    """Base class for a Parser that can parse the outputs produced by a CalcJob process."""

    def __init__(self, node):
        """Construct the Parser instance.

        :param node: the `CalcJobNode` that contains the results of the executed `CalcJob` process.
        """
        from aiida.common.log import AIIDA_LOGGER
        from aiida.orm.utils.log import create_logger_adapter

        self._logger = create_logger_adapter(AIIDA_LOGGER.getChild('parser').getChild(self.__class__.__name__), node)
````
This `create_logger_adapter` should configure the logger such that any messages that get logged are also piped to the `DBLogHandler`. If it doesn't work for you, it would be good if you could open an issue with a MWE.

**Reply by zhubonan** [reply]
I see, thanks!
